dsse/dsse.go

```diff
@@ -61,12 +54,20 @@ type Envelope struct {
 }
 
 type Signature struct {
-	KeyID         string   `json:"keyid"`
-	Signature     []byte   `json:"sig"`
-	Certificate   []byte   `json:"certificate,omitempty"`
-	Intermediates [][]byte `json:"intermediates,omitempty"`
+	KeyID         string               `json:"keyid"`
+	Signature     []byte               `json:"sig"`
+	Certificate   []byte               `json:"certificate,omitempty"`
+	Intermediates [][]byte             `json:"intermediates,omitempty"`
+	Timestamps    []SignatureTimestamp `json:"timestamps,omitempty"`
+}
+
+type SignatureTimestampType string
+
+const TimestampRFC3161 SignatureTimestampType = "tsp"
 
-	trustedTime time.Time
+type SignatureTimestamp struct {
+	Type SignatureTimestampType `json:"type"`
+	Data []byte                 `json:"data"`
```

Comment:
How is this data encoded?

Reply:
Base64. Go json marshals all byte arrays to base64.

Reply:
As far as what data actually is that is an implementation detail of the timestamper being used.

In the case of TSP it’s an ASN1 encoded Timestamp Request object.